python/pyspark/mllib/stat/_statistics.py

```diff
@@ -168,12 +190,34 @@ def corr(x, y=None, method=None):
             raise TypeError("Use 'method=' to specify method name.")
 
         if not y:
-            return callMLlibFunc("corr", x.map(_convert_to_vector), method).toArray()
+            return cast(
+                JavaObject, callMLlibFunc("corr", x.map(_convert_to_vector), method)
+            ).toArray()
         else:
-            return callMLlibFunc("corr", x.map(float), y.map(float), method)
+            return cast(
```

Comment:
`callMLlibFunc` return `Union[JavaObject, bytearray, bytes]` but we expect `Union[float, Matrix]`